File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/equipment/model/dto/response/EquipmentResponseDto.java
package com.junyss.graphqltest.api.equipment.model.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EquipmentResponseDto {
	private String equipmentId;
	private String usedBy;
	private Integer count;
	private String newOrUsed;
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/equipment/model/dto/response/EquipmentResponseDto.kt
package com.junyharangstudy.kotlingraphqltest.api.equipment.model.dto.response

import com.junyharangstudy.kotlingraphqltest.api.equipment.model.entity.Equipment

class EquipmentResponseDto(
    val equipmentId: String,
    val usedBy: String,
    val count: Int,
    val newOrUsed: String
) {
    companion object {
        fun equipmentToDto(equipment: Equipment): EquipmentResponseDto {
            return EquipmentResponseDto(
                equipmentId = equipment.equipmentId,
                usedBy = equipment.usedBy,
                count = equipment.count,
                newOrUsed = equipment.newOrUsed)
        }
    }
}